### PR TITLE
Reconstructive surgery for moth wings

### PR DESCRIPTION
**Pull request by Maurukas**
## About The Pull Request

This adds a new advanced surgery which allows medbay to repair burned off moth wings with the use of synthflesh. It modifies the relevant tech node and medical designs to include this under the appropriate research. It also slightly modifies the mothmen species, as wings being burned off currently destroys that genetic information.

## Why It's Good For The Game

Currently burned off moth wings are the only mutant body part which cannot be repaired or reattached by medbay. In fact they cannot be restored by any in-game means (including magical healing and aheals) other than variable editing. This is something I have heard requested a number of times, and appears to be a fairly long standing issue.

Moth wings confer minor in game benefits and are part of a character's visual identity. At the price of time (and the cost of synthflesh) players now have the option of having an otherwise permanent injury fixed.

## Changelog
:cl:
add: Added reconstructive surgery for moth wings, requiring synthflesh
bugfix: Magical healing and aheals now restore moth wings
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
what if a moth has roundstart burnt off wings? can you make it create default wings if it doesnt find any

**Comment by Maurukas** [comment]
Currently in this PR if you pick the burnt wings in your preferences, it makes you ineligible for this surgery as /datum/species/moth/handle_fire won't apply to you (it checks to see if your wings are already burned). Reconstructing round start burnt wings to 'plain' should be pretty minor though.

**Comment by Jared-Fogle** [comment]
Long term it would be cooler if moth wings were their own organs, so you could recreate this without research by getting a copy of yourself from genetics and transplanting the wings.

File: code/modules/surgery/advanced/wingreconstruction.dm [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+/datum/surgery_step/wing_reconstruction/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
+	var/mob/living/carbon/human/H = target
+	display_results(user, target, "<span class='notice'>You succeed in reconstructing [target]'s wings.</span>",
+		"<span class='notice'>[user] successfully reconstructs [target]'s wings!</span>",
+		"<span class='notice'>[user] completes the surgery on [target]'s wings.</span>")
+	if(H.dna.features["original_moth_wings"] != null)
+		H.dna.features["moth_wings"] = H.dna.features["original_moth_wings"]
+	if(H.dna.features["original_moth_wings"] == null)
+		H.dna.features["moth_wings"] = "Plain"
```
Comment (Jared-Fogle) [inline code/modules/surgery/advanced/wingreconstruction.dm:38]:
```suggestion
	H.dna.features["moth_wings"] = H.dna.features["original_moth_wings"] || "Plain"
```

Comment (Jared-Fogle) [reply code/modules/surgery/advanced/wingreconstruction.dm:38]: Not sure if this is meant to be the "Burnt Off" check, but are you sure `original_moth_wings` isn't just "Burnt Off"?

File: code/modules/surgery/advanced/wingreconstruction.dm [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+				/datum/surgery_step/clamp_bleeders,
+				/datum/surgery_step/wing_reconstruction)
+	possible_locs = list(BODY_ZONE_CHEST)
+	target_mobtypes = list(/mob/living/carbon/human)
+
+/datum/surgery/advanced/wing_reconstruction/can_start(mob/user, mob/living/carbon/target)
+	if(!..())
+		return FALSE
+	if(target.dna.features["moth_wings"] == "Burnt Off")
+		return TRUE
```
Comment (Jared-Fogle) [inline code/modules/surgery/advanced/wingreconstruction.dm:15]:
```suggestion
	return ..() && target.dna.features["moth_wings"] == "Burnt Off"
```

**Comment by ATH1909** [comment]
good PR!

can you make staves of healing restore them too?

File: code/modules/surgery/advanced/wingreconstruction.dm (hunk 1 above)
Comment (Maurukas) [reply code/modules/surgery/advanced/wingreconstruction.dm:38]: `original_moth_wings` should always be null, unless `handle_fire` inserts it into that list when it overwrites `moth_wings`. This proc won't do this if the wings already burnt off, which happens if you pick roundstart burnt wings.

**Comment by zombiegums** [comment]
Thank you for this. I always thought it odd that there was no way to fix tattered wings.

File: code/modules/surgery/advanced/wingreconstruction.dm [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		"<span class='notice'>[user] begins to perform surgery on [target]'s charred wing membranes.</span>")
+
+/datum/surgery_step/wing_reconstruction/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
+	var/mob/living/carbon/human/H = target
+	display_results(user, target, "<span class='notice'>You succeed in reconstructing [target]'s wings.</span>",
+		"<span class='notice'>[user] successfully reconstructs [target]'s wings!</span>",
+		"<span class='notice'>[user] completes the surgery on [target]'s wings.</span>")
+	if(H.dna.features["original_moth_wings"] != null)
+		H.dna.features["moth_wings"] = H.dna.features["original_moth_wings"]
+	if(H.dna.features["original_moth_wings"] == null)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/surgery/advanced/wingreconstruction.dm:37]: `else` instead?

If the first check didnt go through we know the second one must.

File: code/modules/surgery/advanced/wingreconstruction.dm [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	require_all_chems = FALSE
+	experience_given = MEDICAL_SKILL_ADVANCED
+
+/datum/surgery_step/wing_reconstruction/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	display_results(user, target, "<span class='notice'>You begin to fix [target]'s charred wing membranes...</span>",
+		"<span class='notice'>[user] begins to fix [target]'s charred wing membranes.</span>",
+		"<span class='notice'>[user] begins to perform surgery on [target]'s charred wing membranes.</span>")
+
+/datum/surgery_step/wing_reconstruction/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
+	var/mob/living/carbon/human/H = target
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/surgery/advanced/wingreconstruction.dm:31]: assuming!

File: code/modules/surgery/advanced/wingreconstruction.dm (hunk 3 above)
Comment (Jared-Fogle) [reply code/modules/surgery/advanced/wingreconstruction.dm:37]: I made a comment that's less cluttered than an if condition above if you didn't see.

**Comment by Jack7D1** [comment]
Speedmerge

**Comment by RaveRadbury** [comment]
Wasn't this a deliberate design choice?

File: code/modules/surgery/advanced/wingreconstruction.dm (hunk 4 above)
Comment (Maurukas) [reply code/modules/surgery/advanced/wingreconstruction.dm:31]: Pardon, not quite following. Do I need a check for ishuman/ismoth here?

File: code/modules/surgery/advanced/wingreconstruction.dm (hunk 3 above)
Comment (Maurukas) [reply code/modules/surgery/advanced/wingreconstruction.dm:34]: Accepted Jared-Fogle's suggestion above.

File: code/modules/surgery/advanced/wingreconstruction.dm (hunk 1 above)
Comment (Jared-Fogle) [reply code/modules/surgery/advanced/wingreconstruction.dm]: The code suggestion still stands BTW, the question was just a followup.

File: code/modules/surgery/advanced/wingreconstruction.dm (hunk 4 above)
Comment (Maurukas) [reply code/modules/surgery/advanced/wingreconstruction.dm]: Dealt with. Please be patient I'm special.

**Comment by Maurukas** [comment]
I haven't seen anything that indicates this was a design decision, it just looks like an artifact of how they were implemented. Aheals not working was definitely not intended - it's been fixed before.

**Pull request merged**